### DISCUSSION TITLE
Is there a way to offset/move the arrowhead mark along the line?

### DISCUSSION THREAD

**Discussion by Andrew15-5** [cetz-package/cetz#161] [Q&A] [answered]
I want to add the ability to use arrows in my WIP flowchart library. But apparently the tip of the arrow is longer than the line length itself. And the arrow only looks good with `thickness` of `0.1pt`! Any more than that looks ugly.

<details><summary>0.1pt</summary>

![image](https://github.com/johannes-wolf/typst-canvas/assets/37143421/7042b70b-6910-47c1-af80-27c38f622109)

</details>

<details><summary>1pt</summary>

![image](https://github.com/johannes-wolf/typst-canvas/assets/37143421/f5e9ae54-9ae3-4bfc-9f01-3daa4451d50d)

</details>

<details><summary>2pt</summary>

![image](https://github.com/johannes-wolf/typst-canvas/assets/37143421/a64ab901-c8f4-4af0-9220-b278691103ee)

</details>

I have a very messy workaround of checking if the `line.mark.end` is not `none` and add an offset to the block to which the arrow is pointing. In other words, to fix ugly arrows, I need to make my code ugly (the balance of things in nature).

The `mark` only have `start`/`end`/`size` options, but no `offset`/`pos` option (`symbol` doesn't work). In `tikz` there are a lot of things you can tweak with arrowheads. Can I somehow move "back" the arrow just a little bit, so that I don't need to recalculate/modify the coordinates where the drawing (of a flowchart symbol) should start?

**Comment by johannes-wolf** [comment]
You have two options: Fill the arrow but set it's stroke to none, or set the stroke cap style to something different (https://typst.app/docs/reference/visualize/line/#parameters-stroke).

Offsetting the arrow is currently not possible.

**Reply by Andrew15-5** [reply, marked as answer]
If I set arrow's stroke to none, then the line beneath the arrow will be seen near the tip: ![image](https://github.com/johannes-wolf/typst-canvas/assets/37143421/650de3ac-ea26-45d3-aeb1-5e30ba5872d0)

---

`cap` style doesn't change anything, but `join` does. If I set `join` to `"round"`, then given the same thickness, it now doesn't show up from the other side: ![image](https://github.com/johannes-wolf/typst-canvas/assets/37143421/61920fbd-6b93-48d7-86f0-3ca81d51b166)

---

If I instead use default `join: "miter"` and change the value of `miter-limit` to `2` (from default `4.0`) then it will cut out the tip of the arrow: ![image](https://github.com/johannes-wolf/typst-canvas/assets/37143421/bfc5c145-51cf-493e-a84b-81c2704d00ca)

---

With `miter-limit: 1` the effect will be equal to `join: "bevel"` (all 3 corners are cut/clipped): ![image](https://github.com/johannes-wolf/typst-canvas/assets/37143421/1b553a1e-b711-4685-85be-45a50914c664)

---

I've checked: the `cap` only changes style of the non-`fill` arrow (`join` & `miter-limit` will work).

Note: this isn't a 100% perfect solution, because the arrow tip itself is never seen with thickness more than ~`0.1pt`.

**Reply by johannes-wolf** [reply]
Oh yes. I always mix up miter vs cap… great overview!
I tested this with TikZ and it gets it correct, so I will create a ticket: #162.
